### DISCUSSION TITLE
How to use with arrow-flight datasets?

### DISCUSSION THREAD

**Discussion by balshetzer** [duckdb/duckdb#8646] [Q&A]
We have datasets behind an arrow-flight based API. I know that I can create a stream in python and use that as a table in DuckDB. But this doesn't allow DuckDB to do row or column based filtering. 

Is there a way to give DuckDB access to an arrow-flight based dataset with all its capabilities? 

Generally, is there a generic interface one can implement to give DuckDB access to data with the ability to perform optimally?

**Comment by rustyconover** [comment]
Tune into DuckCon 6 on the 2025-01-31, where I'll be talking about a new extension called Airport that integrates DuckDB with Arrow Flight.

https://duckdb.org/events/2025/01/31/duckcon6/

**Reply by dazzleduck-web** [reply]
Looking forward to it. Do you mind sharing the WIP. I might be able to contribute to it. Thanks

**Reply by rustyconover** [reply]
Here is the current repo: https://github.com/Query-farm/duckdb-airport-extension

Expect more docs and examples soon.

**Reply by dazzleduck-web** [reply]
Thanks ...I will explore it soon
